File: 25 Coding Challenge II/JavaScript/Challenge28.js
// Challege 28: Calculate the sum of first 100 prime numbers

// function prints the first nPrime numbers
function printPrime(nPrime){
    let n = 0;
    let i = 2;
    let sum = 0;

    while (n < nPrime){
        if (isPrime(i)){
            sum += i;
            
        }
        n++;
        i++;
    }

    return sum;
}

// function determines if a number is prime
function isPrime(n){
    if (n < 2){
        return false;
    }
    if (n == 2){
        return true;
    }

    let maxDiv = Math.sqrt(n);

    for (let i = 2; i <= maxDiv; i++){
        if (n % i == 0){
            return false;
        }
    }

    return true;
}


// Driver method
console.log("The sum of the first 100 prime number", printPrime(100));

File: 25 Coding Challenge II/Python/Challenge28.py
"""
Challenge 28: calculate the sum of the first 100 prime numbers
"""


# Returns true is a number is prime
import math


def isPrime(n) -> bool:
    if n < 2:
        return False
    if n == 2:
        return True

    maxDiv = math.sqrt(n)
    i = 2
    while i <= maxDiv:
        if n % 1 == 0:
            return False
        i += 1
    return True


def printPrime(nPrimes):
    n = 0
    i = 2

    while n < nPrimes:
        if isPrime(i):
            print(n, "--->", i)
            n += 1
        i += 1


printPrime(100)
